packages/web-client/tests/acceptance/pay-test.ts

```diff
@@ -164,6 +169,38 @@ module('Acceptance | pay', function (hooks) {
     assert.dom(PAYMENT_URL).containsText(expectedUrl);
   });
 
+  test('it rounds SPEND up to the min SPEND amount', async function (assert) {
+    await visit(
+      `/pay/${network}/${merchantSafe.address}?amount=${lessThanMinSpendAmount}&currency=${spendSymbol}`
+    );
+    await waitFor(MERCHANT);
+
+    assert.dom(MERCHANT).hasAttribute('data-test-merchant', merchantName);
+    assert
+      .dom(MERCHANT_LOGO)
+      .hasAttribute(
+        'data-test-merchant-logo-background',
+        merchantInfoBackground
+      )
+      .hasAttribute(
+        'data-test-merchant-logo-text-color',
+        merchantInfoTextColor
+      );
```

Comment:
Merchant details should be the same for each test where we don't fail to fetch merchant safe/display information. Would you use a `afterEach`/`beforeEach` to assert for each case, or simply test it once?

Reply:
Removed some assertions on merchant display, leaving the integration test for `merchant-payment-request-card` to cover any conditional rendering changes in the future.

Reply:
sorry, I saw this in email and then failed to follow up on this page. I would make a dedicated test for this and only make the assertions there, I try to avoid assertions in hooks. The integration test could be fine alone, though I do like to have a bit of redundancy, in this case to make sure the merchant address survives the transition from URL parameter to component invocation.